src/SVTBarrelTracker_geo.cpp

```diff
@@ -0,0 +1,324 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2022 Whitney Armstrong, Jonathan Witte, Shujie Li
+
+/** Curved Silicon Vertex Tracker Barrel with RSU.
+ *
+ * - Designed to process "vertex_barrel_curved.xml"
+ *
+ * - Derived from "BarrelTrackerWithFrame_geo.cpp".
+ * - Build-in RSU structure with 12 tiles and inactive areas
+ *
+ * \code
+ * \endcode
+ *
+ *
+ * @author Whitney Armstrong, Jonathan Witte, Shujie Li
+ */
+
+#include "DD4hep/DetFactoryHelper.h"
+#include "DD4hep/Printout.h"
+#include "DD4hep/Shapes.h"
+#include "DDRec/DetectorData.h"
+#include "DDRec/Surface.h"
+#include "XML/Layering.h"
+#include "XML/Utilities.h"
+#include <array>
+#include "DD4hepDetectorHelper.h"
+
+using namespace std;
+using namespace dd4hep;
+using namespace dd4hep::rec;
+using namespace dd4hep::detail;
+
+static Ref_t create_SVTBarrelTracker(Detector& description, xml_h e, SensitiveDetector sens) {
+
+  xml_det_t x_det = e;
+  Material air    = description.air();
+  int det_id      = x_det.id();
+  string det_name = x_det.nameStr();
+  DetElement sdet(det_name, det_id);
+
+  typedef vector<PlacedVolume> Placements;
+  map<string, Volume> volumes;
+  map<string, Placements> sensitives;
+  map<string, std::vector<VolPlane>> volplane_surfaces;
+  map<string, std::vector<double>> module_length;
+
+  PlacedVolume pv, pv_frame;
+
+  // Set detector type flag
+  dd4hep::xml::setDetectorTypeFlag(x_det, sdet);
+  auto& params = DD4hepDetectorHelper::ensureExtension<dd4hep::rec::VariantParameters>(sdet);
+
+  // Add the volume boundary material if configured
+  for (xml_coll_t bmat(x_det, _Unicode(boundary_material)); bmat; ++bmat) {
+    xml_comp_t x_boundary_material = bmat;
+    DD4hepDetectorHelper::xmlToProtoSurfaceMaterial(x_boundary_material, params,
+                                                    "boundary_material");
+  }
+
+  Assembly assembly(det_name);
+
+  sens.setType("tracker");
+
```

Comment:
I noticed you removed the support and frame sections here. Did you not want to keep the support for this plugin?